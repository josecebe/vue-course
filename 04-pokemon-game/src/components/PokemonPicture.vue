<template>
    <div class="pokemon-container">
        <img alt="Pokémon" class="hidden-pokemon" data-test-id="hidden-pokemon"
            :src="imgSrc" />

        <img alt="Pokémon" class="fade-in" data-test-id="visible-pokemon"
            v-if="showPokemon"
            :src="imgSrc" />
    </div>
</template>

<script>
export default {
    name: 'PokemonPicture',

    props: {
        pokemonId: {
            type: Number,
            required: true,
        },
        showPokemon: {
            type: Boolean,
            required: true,
            default: false,
        }
    },

    computed: {
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`;
        }
    }
}
</script>

<style scoped>
.pokemon-container {
    height: 200px;
}
img {
    height: 200px;
    position: absolute;
    width: 100vw;
    margin-left: -50vw;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}
</style>
