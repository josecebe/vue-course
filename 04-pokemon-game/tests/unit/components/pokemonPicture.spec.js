import {shallowMount} from '@vue/test-utils';
import PokemonPicture from '@/components/PokemonPicture';

describe('PokemonPicture tests', () => {
    test('Should match the snapshot', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should show the hidden image and image should have the correct URL', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        });

        const hiddenImg = wrapper.find('[data-test-id="hidden-pokemon"]');
        const visibleImg = wrapper.find('[data-test-id="visible-pokemon"]');

        expect(hiddenImg.exists()).toBeTruthy();
        expect(hiddenImg.classes('hidden-pokemon')).toBeTruthy();
        expect(hiddenImg.attributes('src')).toContain('/100.svg');
        expect(visibleImg.exists()).toBeFalsy();
    });

    test('Should show the visible pokemon', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        });

        const visibleImg = wrapper.find('[data-test-id="visible-pokemon"]');

        expect(visibleImg.exists()).toBeTruthy();
        expect(visibleImg.classes('hidden-pokemon')).toBeFalsy();
    });
});
