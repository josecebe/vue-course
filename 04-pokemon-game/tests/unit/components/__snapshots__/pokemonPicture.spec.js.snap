// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture tests Should match the snapshot 1`] = `
<div class="pokemon-container"><img alt="PokÃ©mon" class="hidden-pokemon" data-test-id="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg">
  <!--v-if-->
</div>
`;
